lib/__tests__/standalone-quiet-deprecation-warnings.test.mjs

```diff
@@ -21,33 +21,68 @@ const { jest } = import.meta;
 jest.unstable_mockModule('../rules/block-no-empty/index.mjs', () => ({ default: deprecatedRule }));
 
 it('standalone does not silence deprecation warnings by default', async () => {
+	const mock = jest.spyOn(process, 'emitWarning').mockImplementation(() => {});
 	const config = {
 		rules: {
 			'block-no-empty': true,
 		},
 	};
 
 	const { results } = await standalone({ code: 'a { color: #FFF }', config });
+	const text = 'The "block-no-empty" rule is deprecated.';
 
 	expect(results[0].deprecations).toHaveLength(1);
+	expect(results[0].deprecations[0].text).toBe(text);
 	expect(results[0].warnings).toHaveLength(1);
+	expect(mock).toHaveBeenCalledWith(text, {
+		code: 'stylelint:006',
+		detail:
+			'Please be aware that the "block-no-empty" rule will soon be either removed or renamed.',
+		type: 'DeprecationWarning',
+	});
+
+	mock.mockClear();
 });
 
-it('standalone silences deprecation warnings when passed --quiet-deprecation-warnings', async () => {
-	const config = {
-		rules: {
-			'block-no-empty': true,
-		},
-	};
+describe('flags', () => {
+	it('standalone silences deprecation warnings when passed --quiet-deprecation-warnings', async () => {
+		const config = {
+			rules: {
+				'block-no-empty': true,
+			},
+		};
+
+		const { results } = await standalone({
+			code: 'a { color: #FFF }',
+			config,
+			quietDeprecationWarnings: true,
+		});
 
-	const { results } = await standalone({
-		code: 'a { color: #FFF }',
-		config,
-		quietDeprecationWarnings: true,
+		expect(results[0].deprecations).toHaveLength(0);
+		expect(results[0].warnings).toHaveLength(1);
 	});
 
-	expect(results[0].deprecations).toHaveLength(0);
-	expect(results[0].warnings).toHaveLength(1);
+	it('standalone silences deprecation warnings when passed --no-deprecation', async () => {
+		process.noDeprecation = true;
+
+		const config = {
+			rules: {
+				'block-no-empty': true,
+			},
+		};
+
+		const { results } = await standalone({
+			code: 'a { color: #FFF }',
+			config,
+		});
+
+		expect(results[0].deprecations).toHaveLength(0);
+		expect(results[0].warnings).toHaveLength(1);
+	});
+
+	afterEach(() => {
+		process.noDeprecation = undefined;
+	});
```

Comment:
[suggest] If doing this more exactly, we should restore the original `process.noDeprecation` like this:

```js
let originalNoDeprecation;

beforeEach(() => { originalNoDeprecation = process.noDeprecation });

afterEach(() => { process.noDeprecation = originalNoDeprecation });
```

Otherwise, using [`jest.replaceProperty`](https://jestjs.io/docs/jest-object#jestreplacepropertyobject-propertykey-value) may be more smart.

Reply:
> [suggest] If doing this more exactly, we should restore the original `process.noDeprecation` like this:
> 
> ```js
> let originalNoDeprecation;
> 
> beforeEach(() => { originalNoDeprecation = process.noDeprecation });
> 
> afterEach(() => { process.noDeprecation = originalNoDeprecation });
> ```

👍 

> Otherwise, using [`jest.replaceProperty`](https://jestjs.io/docs/jest-object#jestreplacepropertyobject-propertykey-value) may be more smart.

This gives me

> Property `noDeprecation` does not exist in the provided object